src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1010,6 +1014,7 @@ private function addCacheSection(ArrayNodeDefinition $rootNode)
                                 ->end()
                                 ->children()
                                     ->arrayNode('adapters')
+                                        ->useAttributeAsKey('')
```

Comment:
I think the issue here is that we try to support both a map and a list for different cases (see the validation few lines above)